multivac/gather_job_data.py

```diff
@@ -7,19 +7,70 @@
 import re
 import sys
 
+from sensors.failures import failure_specs
+
 PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 sys.path.append(PROJECT_DIR)
 
 version_matcher = re.compile(r"Current runner version: "
                              r"'(\d*.\d*.\d*)'")
 
 
+# As far as the failure occurs at the end of the log, let's
+# start to parse the file from the end to speed up the process
+def reverse_readline(filename, buf_size=8192):
+    """An iterator that returns the lines of a file in reverse order"""
+    with open(filename, encoding='utf8') as fh:
```

Comment:
It is hard to understand what the `fh` means

Reply:
will fix in a further PR